### PR TITLE
Added Readmes for all templates

### LINKED ISSUE #63: Generate README per extension
Should document details about the extension, e.g.:

How to try the feature in the running browser app
That stand alone extension have to manually added and how

### PR DESCRIPTION
**Pull request by JonasHelming**
fixed #63

Signed-off-by: Jonas Helming <jhelming@eclipsesource.com>

### CODE REVIEW

File: templates/backend/README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Backend communication example
```
Comment (vince-fugnitto) [inline templates/backend/README.md:1]: Do you mind if we update the titles of each readme with title casing?

```suggestion
# Backend Communication Example
```

File: templates/backend/README.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+
+The example extension demonstrates how communicate with backend services. It contains two backend service examples:
```
Comment (vince-fugnitto) [inline templates/backend/README.md:3]:
```suggestion
The example extension demonstrates how to communicate with backend services. 
It contains the following two backend service examples:
```

File: templates/backend/README.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+* `HelloBackendService`: Can be called by the client to create a "Hello World" string. The client provides a name as a parameter, the server returns the name prefixed with "Hello", e.g. "World" as a parameter returns "Hello World".
```
Comment (vince-fugnitto) [inline templates/backend/README.md:4]: No strong opinion with this suggestion, just thought it read a little easier.

```suggestion
* `HelloBackendService`: Can be called by the client to create a "Hello World" string. The client provides a name as a parameter, and the server returns the name prefixed with "Hello" (ex: "World" as a parameter returns "Hello World").
```

File: templates/labelprovider/README.md [hunk 4]
```diff
@@ -0,0 +1,8 @@
+# Example label provider
+
+The example extension demonstrates how to contribute a custom label provider customizing the label and icon for specific files (using the ".my" file extension)
```
Comment (vince-fugnitto) [inline templates/labelprovider/README.md:3]: Minor suggestion, no strong opinion either:

```suggestion
The example extension demonstrates how to contribute a custom label provider in order to customize the `label` and `icon` when displaying specific file types (using the ".my" file extension as an example).
```

File: templates/widget/README.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
+# Example widget
+
+The example extension demonstrates how to contribute a custom widget (i.e. a view or editor) to Eclipse Theia.
+
+## How to use the widget example
+
+In the running application, open the widget using the menu "View" => "<%= params.extensionPrefix %> Widget"
```
Comment (vince-fugnitto) [inline templates/widget/README.md:7]: Did you confirm that we are able to substitute the template with the widget name in markdown?
Else we might go for a simpler initial documentation by using an example for instance.

**Review by vince-fugnitto — COMMENTED** [review]
@JonasHelming thank you for taking care of the issue, I provided some initial feedback, please let me know if you agree/disagree with any of the comments.

File: templates/widget/README.md (hunk 5 above)
Comment (JonasHelming) [reply templates/widget/README.md:7]: Yes, I explicitly tested that!

**Comment by JonasHelming** [comment]
@vince-fugnitto : Thank you for the valuable feedback, accepted and squashed...

**Review by vince-fugnitto — COMMENTED** [review]
@JonasHelming do you mind addressing my previous comment: https://github.com/eclipse-theia/generator-theia-extension/pull/84#discussion_r565581290. Only one of the `readmes` were updated to follow `title casing`.

Everything else looks great :+1:

**Comment by JonasHelming** [comment]
> @JonasHelming do you mind addressing my previous comment: [#84 (comment)](https://github.com/eclipse-theia/generator-theia-extension/pull/84#discussion_r565581290). Only one of the `readmes` were updated to follow `title casing`.
> 
> Everything else looks great

Sure, sorry, missed the comment!

**Comment by JonasHelming** [comment]
(done)

**Review by vince-fugnitto — APPROVED** [review]
The changes look good, thank you!

**Pull request merged**